### PR TITLE
Add peer review email

### PR DESCRIPTION
**Pull request by aoby**
[Task](https://trello.com/c/oYt9bPd3/121-send-email-notification-when-your-submission-has-been-peer-reviewed)

I tested this locally using mailcatcher and a review for http://localhost-studio.code.org:3000/s/alltheplcthings/stage/10/puzzle/7

The email looks like this in mailcatcher:
image

And the link brings me back to the level where I see:
image

### CODE REVIEW

File: dashboard/app/mailers/peer_review_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class PeerReviewMailer < ActionMailer::Base
+  default from: 'Code.org <teacher@code.org>'
+
+  def review_completed_receipt(peer_review)
+    @peer_review = peer_review
+    @submission_url = CDO.studio_url @peer_review.submission_path, CDO.default_scheme
+
+    mail(
+      to: @peer_review.submitter.email,
+      subject: "Your submission for \"#{@peer_review.level.name}\" has been reviewed by a peer"
```
Comment (mehalshah) [inline dashboard/app/mailers/peer_review_mailer.rb]: Maybe say "has been reviewed." Some feedback will be from instructors, not peers

File: dashboard/app/models/peer_review.rb [hunk 2]
```diff
@@ -52,6 +54,11 @@ def add_status_to_audit_trail
     append_audit_trail "REVIEWED by user id #{reviewer_id} as #{status}"
   end
 
+  after_save :send_review_completed_mail, if: :status_changed?
+  def send_review_completed_mail
+    PeerReviewMailer.review_completed_receipt(self).deliver_now
```
Comment (mehalshah) [inline dashboard/app/models/peer_review.rb]: Don't send email if the review has been escalated.

File: dashboard/app/views/peer_review_mailer/review_completed_receipt.html.haml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+%p
+  Hi
```
Comment (mehalshah) [inline dashboard/app/views/peer_review_mailer/review_completed_receipt.html.haml]: Just my opinion but I'm more of a Hello person than a hi person. Also comma after name is standard I believe.

File: dashboard/test/models/peer_review_test.rb [hunk 4]
```diff
@@ -525,6 +527,24 @@ class PeerReviewTest < ActiveSupport::TestCase
...
+    assert_equal "/levels/#{standalone_level.id}", peer_review_with_standalone_level.submission_path
+  end
+
+  test 'status change triggers review email' do
+    peer_review = create :peer_review
+    PeerReviewMailer.expects(:review_completed_receipt).with(peer_review).returns(stub(:deliver_now))
+
+    peer_review.update!(status: :accepted)
+  end
+
```
Comment (mehalshah) [inline dashboard/test/models/peer_review_test.rb]: One additional test to make sure that escalating does not trigger email

File: dashboard/app/models/peer_review.rb (hunk 2 above)
Comment (aoby) [reply dashboard/app/models/peer_review.rb]: done - now only send when it changes and is either accepted or rejected

**Pull request merged**